doc/vivify.txt

```diff
@@ -0,0 +1,46 @@
+*vivify.txt*        Markdown preview on the browser
+
+
+INTRODUCTION      *vivify*
+
+Connects (Neo)Vim to the Vivify Markdown preview tool.
+View and live-sync Markdown files while editing.
```

Comment:
Yes I was thinking about that too, but couldn't decide should it be worded like:

* "Vivify can also view other filetypes like yadayada"
* Don't call Vivify a Markdown preview tool, but a general viewer and perhaps even a file browser

Also this will be especially relevant if the Jupytext-to-notebook workflow will become as good a use case as the md editing

Do you have a suggestion? Also feel free to make a commit on this PR

Reply:
Oh I see now, jupyter notebook thing would be a separate vim plugin

Reply:
But lets not get caught up with that until it's relevant okay 😄

Reply:
> Do you have a suggestion? Also feel free to make a commit on this PR

So far not really, it's complicated. We can also just leave it like this and then maybe keep talking about it on the discussion thread over on the main repo.

> Oh I see now, jupyter notebook thing would be a separate vim plugin

Yes, and actually, the Jupyter thing will make the whole ecosystem bigger and more complex ^^

```plain
         Markdown -> vivify.vim
       /
Vivify 

       \
         Notebooks -> Jupyviv -> jupyviv.vim
```

So there will be two new additions:

1. Jupyviv, which bridges the gap between the Notebook JSON format and plain-text editors while managing the Jupyter Kernels and providing an interface that is simple to integrate with a plugin for any editor. This is the more complex system.
2. jupyviv.vim, which is the equivalent of vivify.vim but for Jupyviv, i.e. the integration of said interface. This should be very simple.

This is of course as out of topic as it gets here but I wanted to tell you about it already ^^ also maybe so we can start thinking about how we want to document this whole bigger picture setup.

> But lets not get caught up with that until it's relevant okay 😄

Just as an update: Jupyviv is pretty much done. I will wait to publish it until I finish jupyviv.vim and have tested that everything works well in the "real world" ^^

Reply:
Soooo long story short, should we just merge this as it is for now?

Reply:
Yes, in case you're fine with not explaining what happens when using :Vivify on a directory buffer or for example a .js file, but if someone tries, the behavior is real nice and you can mostly figure out what's happening (maybe as a pleasant surprise) 😄

Reply:
Yes, I guess it's fine ^^ People who want to use this have to go to the main repo and install Vivify anyways so they will have to close their eyes very hard to not read that Vivify does more than just Markdown haha